pkg/costmodel/allocation.go

```diff
@@ -889,7 +895,10 @@ func applyRAMBytesUsedMax(podMap map[podKey]*Pod, resRAMBytesUsedMax []*prom.Que
 	}
 }
 
-func applyGPUsRequested(podMap map[podKey]*Pod, resGPUsRequested []*prom.QueryResult) {
+func applyGPUsRequested(podMap map[podKey]*Pod, resGPUsRequested []*prom.QueryResult, resGPUsAllocated []*prom.QueryResult) {
+	if len(resGPUsAllocated) > 0 { // Use the new query, when it's become available in a window
+		resGPUsRequested = resGPUsAllocated
+	}
```

Comment:
This is confusing to read. Should we name it `applyGPUsAllocated`, so that it's more clear that we're trying to figure out allocation (not request) and defaulting to request if we don't have allocation? (If that is, in fact, what we're saying here.)